Material

Gamebot2 opened the pull request:
Lets see how this goes

matthew-zhang-306 committed:
Beginning AngularJS Material

Split this on a new branch because I am unsure if it will work.

Messing around with AngularJS Material to see if things can look good.

matthew-zhang-306 committed:
Trying to Get Input Looking Good

Apparently md-selects animate super super slowly and have weird color artifacts for some reason

matthew-zhang-306 committed:
Merge branch 'master' into material

matthew-zhang-306 committed:
Patched odd filter bug

That was weird

matthew-zhang-306 committed:
Delete Commit

matthew-zhang-306 commented:
This branch is not ready to be merged yet lol

Although the longer we wait the worse the conflicts will be :/

matthew-zhang-306 committed:
Merge branch 'master' into material

matthew-zhang-306 committed:
Trying to do Material

Taking up this side project again, so far not much progress but we'll see

matthew-zhang-306 committed:
Remove merge artifacts

Whoops they were still there in the Java somehow

matthew-zhang-306 committed:
More Material Ventures

OK some things are starting to make more sense but not really.

Temporarily disabled a global fade-in transition in the CSS because that was conflicting with Material, as I found out

matthew-zhang-306 committed:
Material w/ Student Data

Getting there. Also why do the ink ripples work???

matthew-zhang-306 committed:
InsertRecord + Login Material

The ink ripples magically fixed themselves, yay

Messed around with new directives and got some stuff working a lot better than they did

matthew-zhang-306 committed:
I Understand Now

The title is false and has nothing to do with this commit and the person who wrote the title should be fired

matthew-zhang-306 committed:
EditRecord Material Update

OK that one was more difficult than I expected

I tried to merge the record editing controllers because they share a lot of functionality

Turns out md-select doesn't quite call ng-change in the same way that select does, so issues arose

And I found out that the classes Master and Record, while being very similar, used different variable names for some things, so I changed those things

Turns out those changes broke a few more things, so I don't entirely know if this commit is safe to use - for now assume it's not

matthew-zhang-306 committed:
Create updateStudentCtrl.js

So that thing I did with the record controllers? Went ahead and did it for the student controllers also. So that appears to work.

matthew-zhang-306 committed:
Lingering Console Log

Whoops that caused issues time to delete

matthew-zhang-306 committed:
LineChart Material, Attempted

I say "attempted" because I was trying to implement a slider system for picking the month ranges.

Turns out the sliders in AngularJS Material are so underdeveloped that a) there are no range sliders, b) it is impossible to use form verification and ng-messages with them, c) it is really hard to make the text boxes as big as is desired.

I will be reverting back to numbers there as soon as possible, but the other integrations went smoothly.

matthew-zhang-306 committed:
Fix lineChart form

Removing md-slider, because it doesn't work

matthew-zhang-306 committed:
Header + Other Material

These are the last few html pages to get the material upgrade. A merge is coming but I still need to make sure everything is the way I want.

Like icons. I want to add those to the header, actually. You know what, it might be a couple more commits.

matthew-zhang-306 committed:
Icons

Now the toolbar looks great and like it's meant to be that big.

Also did some patches with the styling in other places.
